### ISSUE TITLE
[prefer-template-literal] string concatenation in object literal reports warning on code that should not be changed

### ISSUE THREAD

**Issue by NateRadebaugh** [angular-eslint/angular-eslint#2906]
If we have a string in a TS file, we cannot convert to template literal.

For instance:
```ts

@Component({
  selector: "app-example",
  template: `
    <div [style.--row-height]="itemSize() + 'px'">
        Example
    </div>
`,
})
```

However, this cannot be converted to template literal since it's already inside a backtick

**Comment by reduckted** [comment]
A few things to note:
1. Using a template literal within the template would be OK if the Angular template was a single line and used `'` or `"` to quote the string: ``'<div [style.--row-height]="`${itemSize()}px`">Example</div>'``.
2. Because of the mix of `'` and `"` in the Angular template, you would need to escape one type of those characters.  That throws out the location where the rule reports the problem, resulting in an incorrect fix - the `\` escape character isn't accounted for, so the reported location ends one character too soon. This occurs regardless of whether you escape the `'` or the `"`.
3. Even if you do use backticks to quote the Angular template, you could still use a template literal by escaping the backticks that are used in the template literal. The Angular compiler seems to be fine with this, but the angular-eslint template parser throws an error: ``Parser Error: Unexpected token Lexer Error: Unexpected character [\] at column 1 in expression [\`\${itemSize()}px\`]``

I think the best thing to do would be for this rule to skip inline templates. Since the inline templates have a known file name pattern, this should be easy for the rule to do automatically.

As a quick workaround, you can disable the rule for inline templates by adding this to your ESLint config:
```js
{
  files: ["**/*inline-template-*.component.html"],
  rules: {
    "angular-template/prefer-template-literal": "off",
  },
}
```
